tests/test_decorations.py

```diff
@@ -254,3 +254,144 @@ def test_frame_holes_for_pseudotiled_client(hlwm, x11, frame_bg_transparent):
+    hlwm.attr.theme.active.title_color = RawImage.rgb2string(active_color)
+    hlwm.attr.theme.normal.color = RawImage.rgb2string(normal_color)
+    hlwm.attr.theme.normal.title_color = RawImage.rgb2string(normal_color)
+    hlwm.attr.theme.urgent.color = RawImage.rgb2string(urgent_color)
+    hlwm.attr.theme.urgent.title_color = RawImage.rgb2string(urgent_color)
+
+    hlwm.attr.theme.title_height = 20
+    hlwm.call(['load', '(clients max:0 {})'.format(' '.join(running_clients))])
+    # split twice to make tab area smaller and screenshots faster :-)
+    hlwm.call(['split', 'bottom'])
+    hlwm.call(['split', 'bottom'])
+
+    winhandle = x11.window(running_clients[0])
+    img = x11.decoration_screenshot(winhandle)
+    color_count = img.color_count_dict()
+    assert urgent_color not in color_count
+
+    # make one of the unfocsed tabs urgent
+    x11.make_window_urgent(x11.window(running_clients[2]))
+    img = x11.decoration_screenshot(winhandle)
+    new_color_count = img.color_count_dict()
+    assert urgent_color in new_color_count
+    # there is one 'normal' and one 'urgent' tab, so the colors should
+    # appear similarly often:
+    assert new_color_count[urgent_color] == new_color_count[normal_color]
+    assert new_color_count[normal_color] == color_count[normal_color] / 2
+
+
+def test_decoration_tab_title_update(hlwm, x11):
+    text_color = (212, 189, 140)
+    hlwm.attr.theme.title_color = RawImage.rgb2string(text_color)
+    hlwm.attr.theme.title_height = 20
+    hlwm.call(['set_layout', 'max'])
+    # split twice to make tab area smaller and screenshots faster :-)
+    hlwm.call(['split', 'bottom'])
+    hlwm.call(['split', 'bottom'])
+
+    count = 5
+    win_handles = [x11.create_client()[0] for _ in range(0, count)]
+
+    # empty all window titles:
+    for wh in win_handles:
+        x11.set_window_title(wh, '')
+
+    # focus handle 0:
+    hlwm.call(['jumpto', x11.winid_str(win_handles[0])])
+
+    # take a screenshot, it should not contain the text_color:
+    assert x11.decoration_screenshot(win_handles[0]).color_count(text_color) == 0
+
+    # change the title of an unfocused window:
+    x11.set_window_title(win_handles[2], 'SOMETHING')
+    # this change should now be visible in the tab bar, at least 5 pixels
+    # should have this color now:
+    assert x11.decoration_screenshot(win_handles[0]).color_count(text_color) > 5
+
+
+@pytest.mark.parametrize("running_clients_num", [4])
+def test_decoration_click_changes_tab(hlwm, mouse, running_clients, running_clients_num):
+    hlwm.call(['load', '(clients max:0 {})'.format(' '.join(running_clients))])
+    hlwm.attr.settings.tabbed_max = True
+    hlwm.attr.theme.title_height = 10
+
+    geo = hlwm.attr.clients.focus.decoration_geometry()
+    tabbar_top_left = geo.topleft()
+    tabbar_bottom_right = geo.topleft() + Point(geo.width, hlwm.attr.theme.title_height())
+    print(f"tl {tabbar_top_left}")
```

Comment:
Do you want to keep the print() statements in this test case?

Reply:
No, I forgot to remove them. now they're gone. Thanks!

Reply:
Thanks! But there is still one printing the ratio, can you remove that one, too?

Reply:
sorry; now it's gone :-)

Reply:
Thanks :)